src/payment/models/bank-tx-return/bank-tx-return.service.ts

```diff
@@ -17,20 +13,36 @@ export class BankTxReturnService {
     private readonly bankTxRepo: BankTxRepository,
   ) {}
 
-  async create(bankTxReturn: BankTxReturnInterface): Promise<BankTxReturn> {
-    let entity = await this.bankTxReturnRepo.findOne({ where: { bankTx: { id: bankTxReturn.bankTx.id } } });
+  async create(bankTx: BankTx): Promise<BankTxReturn> {
+    let entity = await this.bankTxReturnRepo.findOne({ where: { bankTx: { id: bankTx.id } } });
     if (entity) throw new BadRequestException('BankTx already used');
 
-    const chargebackBankTx = await this.bankTxRepo.findOne({ where: { id: bankTxReturn.chargebackBankTxId } });
-    if (!chargebackBankTx) throw new BadRequestException('ChargebackBankTx not found');
+    entity = this.bankTxReturnRepo.create({ bankTx });
 
-    entity = await this.bankTxReturnRepo.findOne({ where: { chargebackBankTx: { id: chargebackBankTx.id } } });
-    if (entity) throw new BadRequestException('ChargebackBankTx already used');
+    return await this.bankTxReturnRepo.save(entity);
+  }
 
-    await this.bankTxRepo.update(chargebackBankTx.id, { type: BankTxType.BANK_TX_RETURN_CHARGEBACK });
+  async update(id: number, dto: UpdateBankTxReturnDto): Promise<BankTxReturn> {
+    const entity = await this.bankTxReturnRepo.findOne({ where: { id } });
+    if (!entity) throw new NotFoundException('BankTxReturn not found');
 
-    entity = this.bankTxReturnRepo.create({ ...bankTxReturn, chargebackBankTx });
+    const update = this.bankTxReturnRepo.create(dto);
 
-    return await this.bankTxReturnRepo.save(entity);
+    // chargeback bank tx
+    if (dto.chargebackBankTxId) {
+      update.chargebackBankTx = await this.bankTxRepo.findOne({ where: { id: dto.chargebackBankTxId } });
+      if (!update.chargebackBankTx) throw new BadRequestException('ChargebackBankTx not found');
+
+      const entity = await this.bankTxReturnRepo.findOne({
```

Comment:
I would not name the inner variable equal to the one in the outer context, can cause confusion 🤔